src/librustc/hir/mod.rs

```diff
@@ -2148,10 +2151,10 @@ pub struct TraitRef {
 impl TraitRef {
     /// Gets the `DefId` of the referenced trait. It _must_ actually be a trait or trait alias.
     pub fn trait_def_id(&self) -> DefId {
-        match self.path.def {
-            Def::Trait(did) => did,
-            Def::TraitAlias(did) => did,
-            Def::Err => {
+        match self.path.res {
+            Res::Def(DefKind::Trait, did) => did,
+            Res::Def(DefKind::TraitAlias, did) => did,
```

Comment:
Use a top level or-pattern?

Reply:
Preexisting.
Also, this method (`hir::TraitRef::trait_def_id`) should probably be refactored away, since it stops compilation if it's an error.

Reply:
cc @estebank